From cb3208c26875975619ae883561abdd8b78ddc25a Mon Sep 17 00:00:00 2001
From: Aurelian Zanoschi <aurelian17@gmail.com>
Date: Fri, 22 Jul 2016 23:28:57 +0300
Subject: [PATCH] RF24Mesh Fix cross compilation

In the current shape, the Makefile provided with the RF24Mesh
package cannot be cross compiled. PREFIX was provided as an
environment variable.

Signed-off-by: Aurelian Zanoschi <aurelian17@gmail.com>
---
 Makefile | 29 ++++++++++++++---------------
 1 file changed, 14 insertions(+), 15 deletions(-)

diff --git a/Makefile b/Makefile
index aaf09c2..6e4155a 100644
--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@
 # use make all and make install to install the library 
 # You can change the install directory by editing the LIBDIR line
 #
-PREFIX=/usr/local
+#PREFIX=/usr/local
 
 # Library parameters
 # where to put the lib
@@ -22,22 +22,22 @@ LIBNAME_RFN=$(LIB_RFN).so.1.0
 
 HEADER_DIR=${PREFIX}/include/RF24Mesh
 
-ARCH=armv6zk
-ifeq "$(shell uname -m)" "armv7l"
-ARCH=armv7-a
-endif
+#ARCH=armv6zk
+#ifeq "$(shell uname -m)" "armv7l"
+#ARCH=armv7-a
+#endif
 
 # Detect the Raspberry Pi from cpuinfo
 #Count the matches for BCM2708 or BCM2709 in cpuinfo
-RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2708)
-ifneq "${RPI}" "1"
-RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2709)
-endif
+#RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2708)
+#ifneq "${RPI}" "1"
+#RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2709)
+#endif
 
-ifeq "$(RPI)" "1"
+#ifeq "$(RPI)" "1"
 # The recommended compiler flags for the Raspberry Pi
-CCFLAGS=-Ofast -mfpu=vfp -mfloat-abi=hard -march=$(ARCH) -mtune=arm1176jzf-s -std=c++0x
-endif
+CCFLAGS=${CXXFLAGS} -Ofast -std=c++0x
+#endif
 
 # make all
 # reinstall the library after each recompilation
@@ -45,11 +45,11 @@ all: librf24mesh
 
 # Make the library
 librf24mesh: RF24Mesh.o
-	g++ -shared -Wl,-soname,$@.so.1 ${CCFLAGS} -o ${LIBNAME_RFN} $^ 
+	$(CXX) -shared -Wl,-soname,$@.so.1 ${CCFLAGS} -o ${LIBNAME_RFN} $^ 
 
 # Library parts
 RF24Mesh.o: RF24Mesh.cpp
-	g++ -Wall -fPIC ${CCFLAGS} -c $^
+	$(CXX) -Wall -fPIC ${CCFLAGS} -c $^
 
 # clear build files
 clean:
@@ -65,7 +65,6 @@ install-libs:
 	@install -m 0755 ${LIBNAME_RFN} ${LIBDIR}
 	@ln -sf ${LIBDIR}/${LIBNAME_RFN} ${LIBDIR}/${LIB_RFN}.so.1
 	@ln -sf ${LIBDIR}/${LIBNAME_RFN} ${LIBDIR}/${LIB_RFN}.so
-	@ldconfig
 
 install-headers:
 	@echo "[Installing Headers]"
-- 
1.9.1

